### PR TITLE
Issue #6916: migrate tests to junit5 for package internal

### PR DESCRIPTION
**Pull request by pbludov**
Issue #6916

Tests of the `internal` package are migrated to Junit5, except of `powermock`

### CODE REVIEW

**Comment by romani** [comment]
Pmd is not happy

**Comment by pbludov** [comment]
> Pmd is not happy

done

**Comment by rnveach** [comment]
@romani CI has only 8 items when usually there is 14-15. I am not seeing pitest CI.

File: config/import-control-test.xml [hunk 1]
```diff
@@ -20,7 +20,15 @@
 
   <!-- Till https://github.com/checkstyle/checkstyle/issues/6916 -->
   <subpackage name="internal">
-    <allow pkg="org.junit"/>
+    <subpackage name="powermock">
+      <!--
+        Powermock is not compatible with Junit5
+        till https://github.com/powermock/powermock/issues/830
+        and https://github.com/junit-team/junit5/issues/201
+        Once done, junit-vintage-engine dependency should be removed from pom.xml.
```
[the remark this answers is not in the document]
Comment (pbludov) [reply config/import-control-test.xml]: Good point. I'll create dedicated issue for powermock migration.

[the remark this answers is not in the document]
Comment (pbludov) [reply config/import-control-test.xml]: Done: https://github.com/checkstyle/checkstyle/issues/7368

**Comment by romani** [comment]
rebased to make sure pitest is ok

**Pull request merged**